<?xml version="1.0" encoding="utf-8"?>
<odoo>
    <data noupdate="1">

        <!--Email template -->
        <record id="email_template_attendance_device" model="mail.template">
            <field name="name">Attendance Machine - Send by Email</field>
            <field name="email_from">System</field>
            <field name="subject">Attendance Machine Connection Problem (Ref {{ object.name or 'n/a' }})</field>
            <field name="partner_to">{{ object.user_id.partner_id.id }}</field>
            <field name="model_id" ref="to_attendance_device.model_attendance_device"/>
            <field name="lang">{{ object.user_id.partner_id.lang }}</field>
            <field name="auto_delete" eval="True"/>
            <field name="body_html"><![CDATA[
<div style="font-family: 'Lucida Grande', Ubuntu, Arial, Verdana, sans-serif; font-size: 12px; color: rgb(34, 34, 34); background-color: #FFF; ">

    <p>Hello <t t-out="object.user_id.partner_id.name or ''"/>,</p>

    <p>There are some problems with attendance machine: <t t-out="object.name"/>. </p>

    <p>Please check and troubleshoot.</p>
    <p>Here is the debugging information: <t t-out="ctx.get('error_msg')"/> </p>

    <p>Best Regard,</p>

</div>
            ]]></field>
        </record>

        <record id="email_template_not_safe_to_clear_attendance" model="mail.template">
            <field name="name">Not Safe to clear Attendance Data - Send by Email</field>
            <field name="email_from">System</field>
            <field name="subject">Not Safe to clear Attendance Data (Ref {{ object.name }})</field>
            <field name="partner_to">{{ object.user_id.partner_id.id }}</field>
            <field name="model_id" ref="to_attendance_device.model_attendance_device"/>
            <field name="auto_delete" eval="True"/>
            <field name="lang">{{ object.user_id.partner_id.lang }}</field>
            <field name="body_html"><![CDATA[
<div style="font-family: 'Lucida Grande', Ubuntu, Arial, Verdana, sans-serif; font-size: 12px; color: rgb(34, 34, 34); background-color: #FFF; ">

    <p>Hello <t t-out="object.user_id.partner_id.name or ''"/>,</p>

    <p>The system tried to delete attendance data from the machine <t t-out="object.name"/> but it did not find safe condition to delete since the whole data was not downloaded from the machine into the system.</p>
    <p>Please also ensure that all the machine users have been mapped with System's employees. Otherwise, it is also not safe for attendance data deletion</p>
    <p>Please check and troubleshoot.</p>
    <p>In a case, users are checking in/out while the system is downloading data, this message could appear. In such the case, you can ignore this message.</p>

    <p>Best Regard,</p>

</div>
            ]]></field>
        </record>

        <record id="email_template_error_get_attendance" model="mail.template">
            <field name="name">Error Downloading Attendance Data - Send by Email</field>
            <field name="email_from">System</field>
            <field name="subject">Error Downloading Attendance Data (Ref {{ object.name }})</field>
            <field name="partner_to">{{ object.user_id.partner_id.id }}</field>
            <field name="model_id" ref="to_attendance_device.model_attendance_device"/>
            <field name="auto_delete" eval="True"/>
            <field name="lang">{{ object.user_id.partner_id.lang }}</field>
            <field name="body_html"><![CDATA[
<div>
    <p>Hello <t t-out="object.user_id.partner_id.name or ''"/>,</p>
    <p>The system tried to download attendance data from the machine <t t-out="object.name"/> but failed.</p>
    <p>NOTE:</p>
    <p>The machine might not return its whole attendance data when requested which could cause this error.</p>
    <p>If such the case, please disconnect the machine from the system then download the data with another mean from your local network. After that, reconnect the machine with the system and continue the data synchronization</p>
    <p>Best Regard,</p>
</div>
            ]]></field>
        </record>

        <record id="email_template_unknown_attendance_status_code" model="mail.template">
            <field name="name">An unknown attendance status code was found - Sent by Email</field>
            <field name="email_from">System</field>
            <field name="subject">Unknown Attendance Status Code (Ref {{ object.name }})</field>
            <field name="partner_to">{{ object.user_id.partner_id.id }}</field>
            <field name="model_id" ref="to_attendance_device.model_attendance_device"/>
            <field name="auto_delete" eval="True"/>
            <field name="lang">{{ object.user_id.partner_id.lang }}</field>
            <field name="body_html"><![CDATA[
                <div style="font-family: 'Lucida Grande', Ubuntu, Arial, Verdana, sans-serif; font-size: 12px; color: rgb(34, 34, 34); background-color: #FFF; ">
                    <p>Hello <t t-out="object.user_id.partner_id.name or ''"/>,</p>
                    <p>The system tried to download attendance data from the machine <t t-out="object.name"/> but failed.</p>
                    <p>We found an unknown attendance status code <t t-out="ctx.get('code')"/> from your machine <t t-out="ctx.get('machine_name')"/> but no such code found in the machine's settings in the system. </p>
                    <p>Please check it again in your machine <t t-out="ctx.get('machine_name')"/> and add that attendance code to the setting. In case you want to ignore to load attendance data with this code, go to the machine setting in the system and check 'Ignore Unknown Code'.</p>
                    <p>Best Regard,</p>
                </div>
                ]]>
            </field>
        </record>
    </data>
</odoo>
